src/datadog-metrics.ts

```diff
@@ -1,47 +1,45 @@
-import { combineLatest, from, Observable } from 'rxjs'
+import { from } from 'rxjs'
+import { filter, switchMap } from 'rxjs/operators'
 import * as sourcegraph from 'sourcegraph'
 
 const STATSD_PATTERN = /statsd\.[^\'\"]+\([\'\"]([^\'\"]+)[\'\"]\)*/gi
 
-export function activate(): void {
-    function decorateEditors(editorsToUpdate: sourcegraph.CodeEditor[]): void {
-        for (const editor of editorsToUpdate) {
-            const decorations: sourcegraph.TextDocumentDecoration[] = []
-            for (const [i, line] of editor.document.text!.split('\n').entries()) {
-                let m: RegExpExecArray | null
-                do {
-                    m = STATSD_PATTERN.exec(line)
-                    if (m) {
-                        decorations.push({
-                            range: new sourcegraph.Range(i, 0, i, 0),
-                            isWholeLine: true,
-                            after: {
-                                backgroundColor: '#774b9e',
-                                color: 'rgba(255, 255, 255, 0.8)',
-                                contentText: ' View metric (Datadog) » ',
-                                linkURL: buildUrl(m[1]).toString(),
-                            },
-                        })
-                    }
-                } while (m)
-                STATSD_PATTERN.lastIndex = 0 // reset
+function decorateEditor(editor: sourcegraph.CodeEditor): void {
+    const decorations: sourcegraph.TextDocumentDecoration[] = []
+    const decorationType = sourcegraph.app.createDecorationType()
+    for (const [i, line] of editor.document.text!.split('\n').entries()) {
+        let m: RegExpExecArray | null
+        do {
+            m = STATSD_PATTERN.exec(line)
+            if (m) {
+                decorations.push({
+                    range: new sourcegraph.Range(i, 0, i, 0),
+                    isWholeLine: true,
+                    after: {
+                        backgroundColor: '#774b9e',
+                        color: 'rgba(255, 255, 255, 0.8)',
+                        contentText: ' View metric (Datadog) » ',
+                        linkURL: buildUrl(m[1]).toString(),
+                    },
+                })
             }
-            const decorationType = sourcegraph.app.createDecorationType()
-            setTimeout(() => editor.setDecorations(decorationType, decorations), 200)
-        }
+        } while (m)
+        STATSD_PATTERN.lastIndex = 0 // reset
     }
-    sourcegraph.workspace.onDidOpenTextDocument.subscribe(() =>
-        decorateEditors(sourcegraph.app.activeWindow!.visibleViewComponents)
+    editor.setDecorations(decorationType, decorations)
+}
+
+export function activate(context: sourcegraph.ExtensionContext): void {
+    const activeEditor = from(sourcegraph.app.activeWindowChanges).pipe(
```

Comment:
This will throw if `sourcegraph.app.activeWindowChanges` is undefined (which, again, will only happen on < 3.1 instances). Move it into the `if` block below.